pkg/service/service.go

```diff
@@ -242,6 +247,9 @@ func (s *Service) handleEgress(p *pipeline.Pipeline) {
 		case res := <-result:
 			// recording finished
 			s.sendEgressUpdate(res)
+			if s.handlingRoomComposite.Load() {
```

Comment:
Can also use AtomicBool's [compare and swap](https://pkg.go.dev/go.uber.org/atomic#Bool.CAS) here